### ISSUE TITLE
Strategy for managing schema version references in SQL files

### ISSUE THREAD

**Issue by k-allagbe** [ai-cfia/ailab-datastore#135]
## Description

Our current approach involves hardcoding the schema version (e.g., `"fertiscan_0.0.13"`) directly in SQL files. This method becomes cumbersome when we need to update the schema version, as it requires manual changes across all SQL files. We need to explore better strategies for managing schema references to make version updates more efficient.

**Comment by Francois-Werbrouck** [comment]
I suggest with the new structure to invest time in resetting bytebase

**Comment by k-allagbe** [comment]
> I suggest with the new structure to invest time in resetting bytebase

Any solution that eases development is welcome.

**Comment by Francois-Werbrouck** [comment]
I've hooked both `nachet/db/bytebase` &  `fertiscan/db/bytebase` to their own GitOps.

To make database changes, a developer would create a migration script conforming to the [naming format](https://www.bytebase.com/docs/vcs-integration/create-migration-files/?source=console). After the script is approved and merged into the main branch, Bytebase will automatically create the issue (Bytebase issue which help us keep track and record) to apply the database changes.

The default file name is <<version>>_<<description>> such as 202401010000_create_hello_table.sql. The migration version, 202401010000, dictates the sequence of changes. The version must be **unique** and **contains only numeric characters**

**Comment by Francois-Werbrouck** [comment]
Having the GitOps webhook will enable us to create a new schema file with a runnable script (The previous build of the `schema_<schema number>` should be runnable by bytebase). Bytebase will only fetch new .sql files merging into the main branch. Therefore, minor changes on the current "used" schema after its deployment will still need to be done manually (as I think it is ok)

This way, we only need to change the naming convention of our schema files and when a new schema is ready to be deployed in the database it will be automatically created once it is merged on the main branch. However, we will always need to "hardcode" (or specify) the schema name within the script.

PS: We also need to change the secrets once a new schema is created.

-----

Maybe we should also explore having a schema file being automatically built with all the other folder important .sql. 
EX: in fertiscan it would create a new schema with all the functions within `/get_inspection`, `/new_inspection`, etc. While also including the tables architecture and the default data (DML).

This would allow us to mitigate the need to:
>when we need to update the schema version, it requires manual changes across all SQL files

**Comment by Francois-Werbrouck** [comment]
I've place this issue in review; since it does not have an active development branch or a PR. I guess we can discuss it.

We could also create documentation to help dev maintain this workflow and give out the basic instruction on how it works?

**Comment by k-allagbe** [comment]
> We could also create documentation to help dev maintain this workflow and give out the basic instruction on how it works?

Yes, please. Good PR material.

**Comment by Francois-Werbrouck** [comment]
Linking this [PR ](https://github.com/ai-cfia/dev-rel-docs/pull/136) which is a great start for the documentation of Bytebase
